go/vt/vtgate/semantics/semantic_state.go

```diff
@@ -162,6 +171,7 @@ type (
 		GetForeignKeyChecksState() *bool
 		KeyspaceError(keyspace string) error
 		GetAggregateUDFs() []string
+		FindMirrorRule(tablename sqlparser.TableName) (*vindexes.MirrorRule, string, topodatapb.TabletType, key.Destination, error)
```

Comment:
I think this can only return the mirror rule

Reply:
same comment as here https://github.com/vitessio/vitess/pull/15992#discussion_r1708577877